revised SSO docs (#553)
neo4j/docs-aura: modules/ROOT/pages/platform/security/single-sign-on.adoc

The required basic SSO configuration information can be retrieved from the IdP.
Entering the Discovery URI pre-fills the fields below. 
If this is not known these fields can be completed manually.

.SSO configuration
[.shadow]
image::sso.png[A screenshot of the SSO configuration dialogue,640,480]

.SSO toggles
[.shadow]
image::ssotoggles.png[A screenshot of the SSO toggles (for password/user and Google social log in) after SSO configuration dialogue,640,480]

== Individual instance level SSO configurations available from Support

Support can assist with:

* Role mapping specific to a database instance
* Custom groups claim besides `groups`
* Updating SSO on already running instances

If you require support assistance, visit link:https://support.neo4j.com/[Customer Support] and raise a support ticket including the following information:

. The _Project ID_ of the projects you want to use SSO for.
See xref:platform/user-management.adoc#_projects[Projects] for more information on how to find your __Project ID__.

. The name of your IdP

> == Authorization vs. Authentication

Reviewer: Parts of this section are more suitable for an introduction.

*Authentication* determines who is allowed in. 
Authentication is handled by Single Sign-On (SSO), which serves as a method for logging in. 

> *Authorization* determines what someone can do when they have logged-in.

Reviewer: No need to explain this to the user, I think we can safely assume that they know the difference if they attempt to set up SSO.

Author: I think it’s really useful to include this somewhere, but perhaps not in this part of the docs.

When I was new to RBAC and SSO, Linda Haviv’s explanation helped me a lot—it clearly drew a dividing line between the two concepts for me. Before this, I used to see RBAC and SSO as one merged concept. Her castle metaphor is especially effective:

https://www.linkedin.com/posts/lindahaviv_lets-learn-about-authentication-vs-authorization-activity-7221535846015868929-ssbq

Access privileges beyond login are managed through roles using Role-Based Access Control (RBAC) an authorization method. 

SSO does not directly grant access to edit project settings—such as changing the project name, managing network access, or modifying instance settings like renaming, pausing, or resuming an instance. 
To determine a user’s access rights to these features, you should use RBAC.

Roles and permissions are managed by RBAC, which decides whether a user can access, view, or modify data within the database instances themselves. 
At this level, role mapping can be utilized to grant users different levels of access based on their roles in their Identity Provider (IdP).

> === RBAC determined roles related to SSO

Reviewer: What does this mean?

Author: Are these roles assigned through SSO configurations and invitations, or are they are assigned separately through RBAC configurations and invitations?

Could it be more like **You can map your IdP groups to Aura roles. The following roles are available:**

It would be cool if we could try it out and see how it works!

|
|===

> Suggestion: An organization's administrator can add Aura as a log-in from a tile in an organization's Apps Dashboard.

Reviewer: ?

|
|===

== Log in flow for users when Organization SSO is enabled

> . A user opens the Aura console and selects Continue with {SSO name}

Reviewer: Could this be a diagram?

. The user authenticates successfully with the IdP
. The user is redirected to the Aura console, to the relevant project

// == We do use ID token login, but they do run out every so often and the user has to re-authenticate, and that is how the Aura Console team decided to build SSO in Aura. 

== Okta SSO configuration step-by-step

. Navigate to your Okta admin portal
. Under Applications click Create App Integration
. For Sign-in method, select OIDC - OpenID Connect
. For Application type, select Web Application
. Enter a name for your application
. Ensure that Authorization Code is selected for Grant type
. Under Sign-in redirect URI’s add https://login.neo4j.com/login/callback as the redirect URI.
. Create an Okta SSO config via console. 
You can do this via the org settings for your org.
. (Optional) Select if you want the SSO config to be applied to org logins, to specific tenants within the org, or both
. Enter a Display Name
. For IdP Type select Okta
. For Client ID enter the Client ID field from the Okta app details page
. For Client Secret enter the client secret from the Okta app details page
. For Discovery URI take the domain from your Okta portal. 

> Should be something like https://dev-29540076-admin.okta.com/ and add .well-known/openid-configuration. 

Reviewer: ```suggestion

```

> Your final url should look something like https://dev-29540076-admin.okta.com/.well-known/openid-configuration 

Reviewer: ```suggestion

```

> . Configure the rest of the SSO config as you’d like

Reviewer: ```suggestion
. Configure the rest of the SSO config to your preferences
```

. Click Create

> . To test Instance SSO, create an instance now in a tenant that has the just created SSO config linked.

Reviewer: ```suggestion
. To test Instance SSO, create an instance in a project that has the newly created SSO config linked.
```

=== Configure groups claim in Okta

You can configure a groups claim in Okta so that your Okta groups are added to your tokens when logging in via SSO. 

> This enables the management of Instance roles via a Role Mapping that is configured on the SSO config.

Reviewer: Why is Role Mapping capitalized?

> For more info see the link:https://developer.okta.com/docs/guides/customize-tokens-groups-claim/main/#add-a-groups-claim-for-the-org-authorization-server[Okta docs]

Reviewer: ```suggestion
For more info see the link:https://developer.okta.com/docs/guides/customize-tokens-groups-claim/main/#add-a-groups-claim-for-the-org-authorization-server[Okta documentation]
```

. In your app details page in Okta, go to the Sign On tab and then edit the OpenID Connect ID Token.

> . Choose Filter as the Groups claim type
> . Choose Matches regex under the Groups claim filter and use .* as the regex

Reviewer: ```suggestion
. Select Filter as the Groups claim type
. Select Matches regex under the Groups claim filter and use .* as the expression.
```
